tests/functional/strategy/test_config.py

```diff
@@ -50,37 +50,89 @@ def test_strategy_setEmergencyExit(strategy, gov, strategist, rando, chain):
 
 
 @pytest.mark.parametrize(
-    "getter,setter,caller,val",
+    "getter,setter,caller,val,guard_allowed, authority_error",
     [
-        ("strategist", "setStrategist", "gov", None),
-        ("rewards", "setRewards", "strategist", None),
-        ("keeper", "setKeeper", "strategist", None),
-        ("minReportDelay", "setMinReportDelay", "strategist", 1000),
-        ("maxReportDelay", "setMaxReportDelay", "strategist", 1000),
-        ("profitFactor", "setProfitFactor", "strategist", 1000),
-        ("debtThreshold", "setDebtThreshold", "strategist", 1000),
+        ("strategist", "setStrategist", "gov", None, True, "!authorized"),
+        ("rewards", "setRewards", "strategist", None, True, "!strategist"),
+        ("rewards", "setRewards", "gov", None, False, "!strategist"),
+        ("keeper", "setKeeper", "strategist", None, True, "!authorized"),
+        ("keeper", "setKeeper", "gov", None, True, "!authorized"),
+        (
+            "minReportDelay",
+            "setMinReportDelay",
+            "strategist",
+            1000,
+            True,
+            "!authorized",
+        ),
+        ("minReportDelay", "setMinReportDelay", "gov", 2000, True, "!authorized"),
+        (
+            "maxReportDelay",
+            "setMaxReportDelay",
+            "strategist",
+            1000,
+            True,
+            "!authorized",
+        ),
+        ("maxReportDelay", "setMaxReportDelay", "gov", 2000, True, "!authorized"),
+        ("profitFactor", "setProfitFactor", "strategist", 1000, True, "!authorized"),
+        ("profitFactor", "setProfitFactor", "gov", 2000, True, "!authorized"),
+        ("debtThreshold", "setDebtThreshold", "strategist", 1000, True, "!authorized"),
+        ("debtThreshold", "setDebtThreshold", "gov", 2000, True, "!authorized"),
     ],
 )
 def test_strategy_setParams(
-    gov, strategist, strategy, rando, getter, setter, caller, val
+    gov,
+    strategist,
+    strategy,
+    rando,
+    getter,
+    setter,
+    caller,
+    val,
+    guard_allowed,
+    authority_error,
 ):
-    if not val:
+    if val is None:
         # Can't access fixtures, so use None to mean any random address
         val = rando
 
     prev_val = getattr(strategy, getter)()
 
-    # Only governance or strategist can set this param
-    with brownie.reverts():
+    # None of these params can be set by a rando
+    with brownie.reverts(authority_error):
         getattr(strategy, setter)(val, {"from": rando})
 
     caller = {"gov": gov, "strategist": strategist}[caller]
 
-    getattr(strategy, setter)(val, {"from": caller})
-    assert getattr(strategy, getter)() == val
+    if guard_allowed:
+        getattr(strategy, setter)(val, {"from": caller})
+        assert getattr(strategy, getter)() == val
+
+        getattr(strategy, setter)(prev_val, {"from": caller})
```

Comment:
Shows that strategist can set, and gov can set too
```suggestion
        getattr(strategy, setter)(prev_val, {"from": gov})
```